walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,72 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import com.google.sps.Event;
+import com.google.sps.MeetingRequest;
+import com.google.sps.TimeRange;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Do not provide any time options if the meeting requested is longer than a whole day. 
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    ArrayList<Event> sortedEvents = new ArrayList<>(events);
+
+    Collections.sort(sortedEvents, new Comparator<Event>() {
+      public int compare(Event a, Event b) {
+        return ((Integer) a.getWhen().start()).compareTo((Integer) b.getWhen().start());
+      }
+    });
```

Comment:
Make the Event class have the comparator and call it here. Similar to what TimeRange class does.